gradle-jdks-setup/src/main/java/org/gradle/wrapper/GradleWrapperMain.java

```diff
@@ -0,0 +1,153 @@
+        String archName = CurrentArch.get().uiName();
+        Path daemonKdkMajorVersionPath = projectHome.resolve("gradle/gradle-daemon-jdk-version");
+        String majorVersion = Files.readString(daemonKdkMajorVersionPath).trim();
+        Path localPathFile = projectHome
+                .resolve("gradle/jdks")
+                .resolve(majorVersion)
+                .resolve(osName)
+                .resolve(archName)
+                .resolve("local-path");
+        String localJdkFileName = Files.readString(localPathFile).trim();
+        Path gradleJdksInstallationDir = getGradleJdksPath();
+        Path jdkInstallationPath = gradleJdksInstallationDir.resolve(localJdkFileName);
+        System.out.println("Setting daemon Java Home to " + jdkInstallationPath.toAbsolutePath());
+        System.setProperty(
+                "org.gradle.java.home", jdkInstallationPath.toAbsolutePath().toString());
+
+        String localPathPattern = String.format("%s/%s/local-path", osName, archName);
+        List<String> toolchainsInstallationPaths = getAllInstallationsPaths(
+                projectHome.resolve("gradle/jdks").toFile(), gradleJdksInstallationDir, localPathPattern);
+        System.out.println("Setting custom toolchains locations to " + toolchainsInstallationPaths);
+        return toolchainsInstallationPaths;
+    }
+
+    private static List<String> getAllInstallationsPaths(
+            File gradleJdkConfigurationDir, Path gradleJdksInstallationDir, String localPathPattern)
+            throws IOException {
+        List<String> paths = new ArrayList<>();
+        File[] files = gradleJdkConfigurationDir.listFiles();
+        if (files == null) {
+            return paths;
+        }
+        for (File file : files) {
+            if (file.isDirectory()) {
+                paths.addAll(getAllInstallationsPaths(file, gradleJdksInstallationDir, localPathPattern));
+            } else if (file.getPath().endsWith(localPathPattern)) {
+                String localJdkFileName =
+                        Files.readString(file.toPath().toAbsolutePath()).trim();
+                paths.add(gradleJdksInstallationDir
+                        .resolve(localJdkFileName)
+                        .toAbsolutePath()
+                        .toString());
+            }
+        }
+        return paths;
+    }
+
+    private static boolean isRunningFromGradlew() {
+        return Boolean.parseBoolean(System.getProperty(RUNNING_FROM_GRADLEW_PROP, "false"));
+    }
+
+    private static Path projectHome() {
+        return wrapperJar().toAbsolutePath().getParent().getParent().getParent();
+    }
+
+    private static Path wrapperJar() {
+        URI location;
+        try {
+            location = GradleWrapperMain.class
+                    .getProtectionDomain()
+                    .getCodeSource()
+                    .getLocation()
+                    .toURI();
+        } catch (URISyntaxException var3) {
```

Comment:
Probably give this a better name than `var3`, even if it's `e` (was this copied from a decompiler?)